javagen/src/main/java/com/azure/autorest/template/ClientMethodTemplate.java

```diff
@@ -464,14 +465,48 @@ protected static void convertClientTypesToWireTypes(JavaBlock function, ClientMe
     }
 
     private static boolean addSpecialHeadersToRequestOptions(JavaBlock function, ClientMethod clientMethod) {
+        // logic only works for DPG, protocol API, on RequestOptions
+
         boolean requestOptionsLocal = false;
-        addSpecialHeadersToLocalVariables(function, clientMethod);
-        if (MethodUtil.isMethodIncludeRepeatableRequestHeaders(clientMethod.getProxyMethod())) {
+
+        final boolean repeatabilityRequestHeaders = MethodUtil.isMethodIncludeRepeatableRequestHeaders(clientMethod.getProxyMethod());
+
+        // optional parameter is in getAllParameters
+        boolean bodyParameterOptional = clientMethod.getProxyMethod().getAllParameters().stream()
+                .anyMatch(p -> p.getRequestParameterLocation() == RequestParameterLocation.BODY
+                        && !p.isConstant() && !p.isFromClient() && !p.isRequired());
+        boolean singleContentType = clientMethod.getProxyMethod().getAllParameters().stream()
+                .noneMatch(p -> p.getRequestParameterLocation() == RequestParameterLocation.HEADER
+                        && HttpHeaderName.CONTENT_TYPE.getCaseInsensitiveName().equalsIgnoreCase(p.getRequestParameterName())
+                        && p.getRawType() instanceof EnumType
+                        && ((EnumType) p.getRawType()).getValues().size() > 1);
```

Comment:
logic here is that if protocol API is multiple content-type, either swagger defines a `content-type` header parameter, or codegen adds a `content-type` parameter, for user to set the content-type.